### PR TITLE
CHAD-9480 Add fallback to group 0

### PR DESCRIPTION
**Pull request by greens**
The ikea motion sensor and the somfy buttons both also had this fallback behavior in their DTHs, even though it is not on-spec.

### CODE REVIEW

File: drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua [hunk 1]
```diff
@@ -63,6 +63,7 @@ local function zdo_binding_table_handler(driver, device, zb_rx)
       return
     end
   end
+  driver:add_hub_to_zigbee_group(0x0000) -- fallback if no binding table entries found
```
Comment (tpmanley) [inline drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:66]: The DTH also added the device to group 0x0000 so I think we also need to do that here.

Comment (greens) [reply drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:66]: you mean this line:

`cmds += zigbee.command(CLUSTER_GROUPS, 0x00, "${zigbee.swapEndianHex(zigbee.convertToHexString(groupAddr, 4))} 00")`

Comment (tpmanley) [reply drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:66]: Yep that's the one

File: drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua [hunk 2]
```diff
@@ -63,6 +64,8 @@ local function zdo_binding_table_handler(driver, device, zb_rx)
...
+  device:send(Groups.commands.AddGroup(device, 0x0000, "00"))
```
Comment (greens) [inline drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:68]: @tpmanley this is how I parsed that command from the DTH, but would love some confirmation.

File: drivers/SmartThings/zigbee-motion-sensor/src/test/test_ikea_motion.lua [hunk 3]
```diff
@@ -68,62 +70,83 @@ test.register_message_test(
...
-          }
-      )
-      test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.motionSensor.motion.active()))
-      test.wait_for_events()
-      test.mock_time.advance_time(180)
-      test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.motionSensor.motion.inactive()))
-    end
+  "ZDO Message handler and adding hub to group",
+  function()
+    local binding_table = mgmt_bind_response.BindingTableListRecord("\x6A\x9D\xC0\xFE\xFF\x5E\xCF\xD0", 0x01, 0x0006, 0x01, 0xB9F2)
```
Comment (greens) [inline drivers/SmartThings/zigbee-motion-sensor/src/test/test_ikea_motion.lua:134]: wanted to add a confirming test here, but not sure how to construct the packet that prompts the fallback

Comment (tpmanley) [reply drivers/SmartThings/zigbee-motion-sensor/src/test/test_ikea_motion.lua:134]: Could set `binding_table_list_count` to 0x00 and not provide any entries

File: drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua (hunk 2 above)
Comment (tpmanley) [reply drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:68]: @cjswedes @varzac , double check me here, but I think for a `CharString` the length is handled automatically so we can just pass an empty string here.

File: drivers/SmartThings/zigbee-button/src/zigbee-multi-button/somfy/init.lua [hunk 4]
```diff
@@ -22,6 +22,7 @@ local function zdo_binding_table_handler(driver, device, zb_rx)
       return
     end
   end
+  driver:add_hub_to_zigbee_group(0x0000) -- fallback if no binding table entries found
```
Comment (tpmanley) [inline drivers/SmartThings/zigbee-button/src/zigbee-multi-button/somfy/init.lua:25]: Looking at the DTH Somfy also needs to add the device to group 0x0000 in this case.

File: drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua (hunk 2 above)
Comment (varzac) [reply drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:68]: Correct, just the value needs to be included

Comment (greens) [reply drivers/SmartThings/zigbee-motion-sensor/src/ikea/init.lua:68]: oh, that `00` in the DTH was just encoding the length of the (empty) group name? Got it.

**Review by tpmanley — APPROVED** [review]

**Pull request merged**